cclient/serialization/json/json_serializer.c

```diff
@@ -33,9 +85,53 @@ void json_get_balances_serialize_request(const serializer_t* const s,
                                          const get_balances_req_t* const obj,
                                          char* out) {}
 
-void json_get_balances_deserialize_response(const serializer_t* const s,
-                                            const char* const obj,
-                                            get_balances_res_t* out) {}
+int json_get_balances_deserialize_response(const serializer_t* const s,
+                                           const char* const obj,
+                                           get_balances_res_t* out) {
+  cJSON* json_obj = cJSON_Parse(obj);
+  cJSON* json_item = NULL;
+
+  if (json_obj == NULL) {
+    cJSON_Delete(json_obj);
+    return EXIT_FAILURE;
+  }
+
+  json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "error");
+  if (cJSON_IsString(json_item) && (json_item->string != NULL)) {
+    cJSON_Delete(json_obj);
+    return EXIT_FAILURE;
+  }
+
+  json_item = cJSON_GetObjectItemCaseSensitive(json_obj, "balances");
+
+  if (cJSON_IsArray(json_item)) {
+    int int_items = cJSON_GetArraySize(json_item);
+    if (int_items <= 0) {
+      cJSON_Delete(json_obj);
+      return EXIT_FAILURE;
+    }
+  }
+
+  int int_cnt = 0;
+  cJSON* current_obj = NULL;
+  cJSON_ArrayForEach(current_obj, json_item) {
```

Comment:
how do you know that `balances.array` is large enough?

Reply:
I don't know the number from the response, the user should allocate a sensible size to store the `balances.array`, if `balances.array` from the response is more then `out->balances.size` which is allocated by a user, only keep the number of `out->balances.size`.